File: archive-application/src/main/java/site/archive/dto/v2/OAuthRegisterRequestDto.java
package site.archive.dto.v2;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import site.archive.domain.user.BaseUser;
import site.archive.domain.user.OAuthProvider;
import site.archive.domain.user.OAuthUser;
import site.archive.domain.user.UserRole;

@Getter
@NoArgsConstructor
@AllArgsConstructor
public class OAuthRegisterRequestDto {

    private OAuthProvider provider;
    private String email;
    private String nickname;

    public BaseUser toUserEntity() {
        return new OAuthUser(this.email,
                             UserRole.GENERAL,
                             this.provider,
                             this.nickname);
    }

}

File: archive-application/src/main/kotlin/site/archive/dto/v2/OAuthRegisterRequestDto.kt
package site.archive.dto.v2

import site.archive.domain.user.BaseUser
import site.archive.domain.user.OAuthProvider
import site.archive.domain.user.OAuthUser
import site.archive.domain.user.UserRole

data class OAuthRegisterRequestDto(
    val provider: OAuthProvider,
    val email: String,
    val nickname: String
) {

    fun toUserEntity(): BaseUser {
        return OAuthUser(email, UserRole.GENERAL, provider, nickname)
    }

}